api/api.go

```diff
@@ -664,3 +668,853 @@ func (f *PublicController) SearchCourse(c *gin.Context) {
+	exams, err := pCtrl.GetRegisteredExamsFromUser(userId)
+	if err != nil {
+		log.Errorf("Unable to get exams from user: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	// Return Status and Data in JSON-Format
+	c.IndentedJSON(http.StatusOK, exams)
+}
+
+func (f *PublicController) GetUnregisteredExamsFromUser(c *gin.Context) {
+	userId, err := f.GetIdFromCookie(c)
+	if err != nil {
+		log.Errorf("Unable to get id from Cookie: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	pCtrl := exam.PublicController{Database: f.Database}
+	exams, err := pCtrl.GetUnregisteredExams(userId)
+	if err != nil {
+		log.Errorf("Unable to get exams: %s", err.Error())
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+	c.IndentedJSON(http.StatusOK, exams)
+}
+
+func (f *PublicController) GetExamsFromCourse(c *gin.Context) {
+	courseId, err := strconv.Atoi(c.Param("id"))
+	if err != nil {
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+	// Fetch Data from Database with Backend function
+	pCtrl := exam.PublicController{Database: f.Database}
+	exams, err := pCtrl.GetExamsFromCourse(courseId)
+	if err != nil {
+		log.Errorf("Unable to get exam: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	// Return Status and Data in JSON-Format
+	c.IndentedJSON(http.StatusOK, exams)
+}
+
+func (f *PublicController) GetAttendedExamsFromUser(c *gin.Context) {
+	// Get given ID from the Context
+	// Convert data type from str to int to use ist as param
+	user_id, err := f.GetIdFromCookie(c)
+	if err != nil {
+		log.Errorf("Unable to get id from Cookie: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	// Fetch Data from Database with Backend function
+	pCtrl := exam.PublicController{Database: f.Database}
+	exams, err := pCtrl.GetAttendedExamsFromUser(user_id)
+	if err != nil {
+		log.Errorf("Unable to get exams from user: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	// Return Status and Data in JSON-Format
+	c.IndentedJSON(http.StatusOK, exams)
+}
+
+func (f *PublicController) GetPassedExamsFromUser(c *gin.Context) {
+	// Get given ID from the Context
+	// Convert data type from str to int to use ist as param
+	user_id, err := f.GetIdFromCookie(c)
+	if err != nil {
+		log.Errorf("Unable to get id from Cookie: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	// Fetch Data from Database with Backend function
+	pCtrl := exam.PublicController{Database: f.Database}
+	exams, err := pCtrl.GetPassedExamsFromUser(user_id)
+	if err != nil {
+		log.Errorf("Unable to get exams from user: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	// Return Status and Data in JSON-Format
+	c.IndentedJSON(http.StatusOK, exams)
+}
+
+func (f *PublicController) GetCreatedFromUser(c *gin.Context) {
```

Comment:
autorisierung fehlt

Reply:
Durch den DB-Zugriff in exam.go wird implizit schon alles geholt, was zur Cookie-id passt. Da ist keine Autorisierung nötig.